client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,691 @@
+            ZIO.logWarning("Received ServerRequest while connecting, ignoring").as(this)
+
+          case StreamEvent.ServerMsg(RequestError(_, _)) =>
+            ZIO.logWarning("Received RequestError while connecting, ignoring").as(this)
+
+          case StreamEvent.ServerMsg(SessionClosed(_, _)) =>
+            ZIO.logWarning("Received SessionClosed while connecting, ignoring").as(this)
+
+          case StreamEvent.Action(ClientAction.Disconnect) =>
+            for {
+              _ <- transport.disconnect().orDie
+            } yield Disconnected
+
+          case StreamEvent.Action(ClientAction.Connect) => 
+            ZIO.logWarning("Received Connect while connecting, ignoring").as(this)
+
+          case StreamEvent.KeepAliveTick =>
+            ZIO.succeed(this)                    
+        }
+      }
+    }
+
+    /** Connecting to resume an EXISTING session (has sessionId).
+      */
+    private case class ConnectingExistingSession(
+        sessionId: SessionId,
+        capabilities: Map[String, String],
+        nonce: Nonce,
+        addresses: List[String],
+        currentAddressIndex: Int,
+        createdAt: Instant,
+        serverRequestTracker: ServerRequestTracker,
+        nextRequestId: RequestIdRef,
+        pendingRequests: PendingRequests
+    ) extends ClientState {
+      override def stateName: String = s"ConnectingExistingSession($sessionId)"
+
+      override def handle(
+          event: StreamEvent,
+          transport: ZmqClientTransport,
+          config: ClientConfig,
+          serverRequestQueue: Queue[ServerRequest]
+      ): UIO[ClientState] = {
+        event match {
+          case StreamEvent.ServerMsg(SessionContinued(responseNonce)) =>
+            if (nonce == responseNonce) {
+              for {
+                _ <- ZIO.logInfo(s"Session continued: $sessionId")
+                now <- Clock.instant
+                // Send all pending requests
+                updatedPending <- pendingRequests.resendAll(transport)
+              } yield Connected(
+                sessionId = sessionId,
+                capabilities = capabilities,
+                createdAt = now,
+                serverRequestTracker = serverRequestTracker,
+                nextRequestId = nextRequestId,
+                pendingRequests = updatedPending,
+                addresses = addresses,
+                currentAddressIndex = currentAddressIndex
+              )
+            } else {
+              ZIO.logWarning("Nonce mismatch, ignoring SessionContinued").as(this)
+            }
+
+          case StreamEvent.ServerMsg(SessionRejected(reason, responseNonce, leaderId)) =>
+            if (nonce == responseNonce) {
+              reason match {
+                case NotLeader =>
+                  val nextIndex = (currentAddressIndex + 1) % addresses.length
+                  val nextAddr = addresses(nextIndex)
+                  for {
+                    _ <- ZIO.logInfo(s"Not leader, trying next: $nextAddr")
```

Comment:
we need to make sure all the connecting log message are consisted.

The log message for the not leader, should be on the old address, so two log messages.
Like "Connecting Existing Session $sessionId to $nextAddr"